src/Symfony/Component/Console/Application.php

```diff
@@ -947,6 +957,10 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             throw $e;
         }
 
+        if ($command instanceof LockableInterface) {
+            $command->release();
```

Comment:
shouldnt it be released before the throw...

Reply:
Your right! Fixed in latest push.